### ISSUE TITLE
new "old" bug  about picture/img  main channel  and clone in one post

### ISSUE THREAD

**Issue by elvecio** [redmatrix/hubzilla#919]
I did  a post with a picture ( the file is in the main  channel hub), but in my clone  , if  I look  the source , I see a new file ( the picture ) in that post. But...where is the picture/file? I don't have that picture,  I don't have nothing. In the clone  just  have  a post without  picture. Is an old bug ( now closed).

**Comment by elvecio** [comment]
I also did the opposite (from the clone to the main) and... the same problem.
In either case, however, I started, with the original post, from the hub with older software version , and  in the hub with  newest version I didn't got the picture ( same post but without the picture).

**Comment by git-marijus** [comment]
It would be kind of you if you mentioned what versions are involved.
Also, if you are sure it's the same bug you have seen a while ago and the bug was closed, it would be better to reopen the bug, or at least post a reference to the issue...

**Comment by elvecio** [comment]
where are the old closed bugs?

**Comment by elvecio** [comment]
Found ... 2 : https://github.com/redmatrix/hubzilla/issues/769 and https://github.com/redmatrix/hubzilla/issues/764

**Comment by zotlabs** [comment]
The old bugs were fixed and have been working well for man months. 

The original and the clone need to have their clocks in sync within +- 3 minutes or file synchronisation will fail. Also if the sending site has to queue the sync packet for any reason (system is heavily loaded, network timeout, etc.) this could also cause a 3 minute delay and invalidate the security token. The queue mechanism was recently changed to automatically queue posts and defer immediate delivery to keep the number of processes under control - as reported by another member. This has probably resulted in these sync packets queuing and missing the 3-minute window of opportunity to initiate a file transfer. 

This is a security measure to prevent token theft/re-use and can probably be eliminated by authenticating to mod_getfile with OWA; which didn't exist when the file synchronisation code was first developed.

**Comment by zotlabs** [comment]
Could not duplicate here and I checked the timestamp issue and found that it is not subject to queue delays, only the time difference between the two servers in question.

I've added OWA as an auth provider for this interface so that in the future the time difference between servers will not be an issue, but I can't change existing installations. So either the clocks need to be synchronised, or you will have issues until both sites are running with an OWA provider at some point in the future.
